uni/lib/redux/action_creators.dart

```diff
@@ -61,14 +62,15 @@ ThunkAction<AppState> reLogin(
       store.dispatch(SetLoginStatusAction(RequestStatus.busy));
       final Session session =
           await NetworkRouter.login(username, password, faculties, true);
-      store.dispatch(SaveLoginDataAction(session));
       if (session.authenticated) {
+        store.dispatch(SaveLoginDataAction(session));
```

Comment:
Move this to line 64.
If the login fails, it returns a session object with authenticated at false.
It makes sense to also saved that.